core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -71,6 +72,11 @@ private[spark] object CallSite {
 private[spark] object Utils extends Logging {
   val random = new Random()
 
+  // Use a common file numbering format which defaults to 5 digits for saving various part files.
+  val numFormatter = NumberFormat.getIntegerInstance()
```

Comment:
If this is used by multiple different classes for formatting the number if seems fine to be in Utils. How about instead we have a function in here that is formatPartitionNumber(int id) which internally uses the number formatter?